tests/purchase_test.ts

```diff
@@ -0,0 +1,39 @@
+import LandingPage from "../pages/landingPage";
+import LoginPage from "../pages/loginPage";
+import ProductPage from "../pages/productPage";
+import CartPage from "../pages/cartPage";
+import CheckoutPage from "../pages/checkoutPage";
+import { User } from "../helpers/interfaces";
+import Products from "../testdata/products";
+
+Feature('Purchase');
+
+let products = new DataTable(['id', 'color', 'size']);
```

Comment:
Store color/size in single item `options` as object: {color: "Black", size: "Large"}, null if no options at all, exclude key, if only other option exists. Then you can pass only single object to your select() method and avoid undefined in DataTable